### ISSUE TITLE
Light up assigned exercises in-context

### ISSUE THREAD

**Issue by rbeezer** [RunestoneInteractive/rs#16]
@oscarlevin wants to have `exercise` of a PreTeXt book, that are actually assigned, to be marked in some way that is clear to the reader at the location of the exercise in the text (in context).  On a Runestone Server, the assignments for a course are created by the instructor and stored.

Analogous to the progress indicator, is there something that can go in every `exercise` that will allow it to "light up" for the reader if, and only if, it is in the database as assigned?

We know the "Runestone ID" of the interior (interactive) Runestone component.  That could be placed in the enclosing PreTeXt `exercise`, `PROJECT-LIKE`, OR `task` somehow (HTML attribute? `data-runestone-id`?) and if that interior component is assigned, then maybe the problem number changes color, gets bigger, is in a box or circle, or something that screams "I am assigned!".

Obviously I'm glossing over the server-side details, but it seems very similar to me to the progress indicator, which is doing a per-student, per-page database query, it seems.

Might be a fiddle to push the Runestone ID outwards to the wrapper, so if probing for it is not hard, then...

**Comment by Alex-Jordan** [comment]
Has this feature (the paperclip) been implemented?

**Comment by bnmnetp** [comment]
It has been done for webwork questions.  Not for Runestone questions.   But it mostly seems redundant for Runestone questions since we restore the feedback, so students know that they have answered the question.  This would be relatively simple for a group of students in @pearcej class to take on.

**Comment by Alex-Jordan** [comment]
I don't see this happening in `portlandcommunitycollege_orcca_fall23`. For instance I have an assignment called "Section 1.1" which includes the exercise numbered 2 in that section of the book (which is a WW problem). But logged in as a student (my jordanstudent account) I do not see any indication in context that this exercise is part of an assignment. (I don't see the paperclip.)

**Comment by bnmnetp** [comment]
No, I have not implemented the paperclip, sorry.  Only that the student has the question right or wrong.

**Comment by Alex-Jordan** [comment]
OK, I'm glad it's not the case that it's implemented and it's a bug that I don't see it :)

Do you have thoughts on how to implement this? I am wondering if the following is too crude: Upon page load, query through all "problem set"s, gathering a list of all id~s of exercises that are in use in some "problem set". Including no doubt many exercise id~s that are not even present on the current page.

Then construct a CSS snippet that styles based on those id's, and insert it into action. I'm wondering if that could be more efficient than visiting all of the exercises on the page and messing with their DOM structure.

Nothing urgent here. Just following up with my thoughts.

**Comment by bnmnetp** [comment]
The paper clip idea seems good in theory, but...

Should a paper clip show for all assignments?  Assignments that are due in the next two weeks?  Some instructors have all of their assignments for the semester queued up on day one. At some point this becomes more confusing because the student sees it is part of an assignment but has no idea of the due date..  If it is part of a review assignment, many teachers would not want students doing the problem ahead of time.

This could add significant load to the servers, and I'm kind of wondering if there is much to gain.  The course page shows the assignments now, which should make it easy for students to click on an assignment and see the problems.

**Comment by Alex-Jordan** [comment]
I think with math courses, it would be a feature that helps a lot. Anecdotally I already had one student plodding along through all 100 or so exercises in context, instead of just doing the selected dozen that were assigned. In that case it's just that 1 in 20 or so students will inevitably have missed/forgotten/in-one-ear-out-the-other'd the guidance for how to do assignments. 

But I understand what you are saying. Perhaps "problem sets" could have more structure? Right now, it looks like they have:

- due date
- visible to students boolean
- allow late submissions boolean
- timed assessment boolean
- time limit for timed assessment
- peer instruction boolean
- show async boolean
- description

I could probably suggest a dozen more useful attributes, based on helpful things I've seen in other platforms. For this discussion, additional dates and a boolean could help address what you are bringing up:

- open date (before this date, students cannot access the assignment, defaults to beginning of time)
- close date (comes on or after due date, and removes access to the assignment, defaults to end of time)
- show in context boolean (between open date and close date, show that paperclip in context for the exercises in this set)

**Comment by bnmnetp** [comment]
Thanks Alex, 

Lets get all the configuration stuff on the table...

To be up front with my bias, I have an allergy to too much configuration, as I think it is the downfall of many systems, and adds complexity and maintenance cost going forward.  But I'm willing to consider all the options.

**Comment by Alex-Jordan** [comment]
Are you asking me to suggest more config options I have seen in other platforms? The ones I think would be relevant for the "paperclip" are the three I mentioned. "Open date" and "close date" also help an instructor prevent the student from being overwhlemed with a huge list of assignments at any one given time.

Other config options I can think of don't seem relevant to this thread, but I could share somewhere. A new issue? I just don't want to confuse me sharing all of them with me lobbying for them in RA.
